### ISSUE TITLE
Interactive Message Filtering Bricked in Latest Update

### ISSUE THREAD

**Issue by awcchungster** [slackapi/node-slack-interactive-messages#88]
### Description

Describe your issue here.

- [X ] bug
- [X ] question

### Bug Report

Filling out the following details about bugs will help us solve your issue sooner.

#### Steps to reproduce:

1. Use block kit found [here](https://api.slack.com/tools/block-kit-builder?blocks=%5B%0A%09%7B%0A%09%09%22type%22%3A%20%22section%22%2C%0A%09%09%22text%22%3A%20%7B%0A%09%09%09%22type%22%3A%20%22mrkdwn%22%2C%0A%09%09%09%22text%22%3A%20%22Escalated%20by%20Gatekeeper%20for%20Approval%20-%20Request%201000085%22%0A%09%09%7D%0A%09%7D%2C%0A%09%7B%0A%09%09%22type%22%3A%20%22divider%22%0A%09%7D%2C%0A%09%7B%0A%09%09%22type%22%3A%20%22section%22%2C%0A%09%09%22fields%22%3A%20%5B%0A%09%09%09%7B%0A%09%09%09%09%22type%22%3A%20%22plain_text%22%2C%0A%09%09%09%09%22text%22%3A%20%22first%20middle%20last%20name%22%2C%0A%09%09%09%09%22emoji%22%3A%20true%0A%09%09%09%7D%2C%0A%09%09%09%7B%0A%09%09%09%09%22type%22%3A%20%22plain_text%22%2C%0A%09%09%09%09%22text%22%3A%20%22email%40gmail.com%22%2C%0A%09%09%09%09%22emoji%22%3A%20true%0A%09%09%09%7D%2C%0A%09%09%09%7B%0A%09%09%09%09%22type%22%3A%20%22plain_text%22%2C%0A%09%09%09%09%22text%22%3A%20%22(111)222-3333%22%2C%0A%09%09%09%09%22emoji%22%3A%20true%0A%09%09%09%7D%2C%0A%09%09%09%7B%0A%09%09%09%09%22type%22%3A%20%22plain_text%22%2C%0A%09%09%09%09%22text%22%3A%20%22%20%22%2C%0A%09%09%09%09%22emoji%22%3A%20true%0A%09%09%09%7D%0A%09%09%5D%0A%09%7D%2C%0A%09%7B%0A%09%09%22type%22%3A%20%22section%22%2C%0A%09%09%22fields%22%3A%20%5B%0A%09%09%09%7B%0A%09%09%09%09%22type%22%3A%20%22plain_text%22%2C%0A%09%09%09%09%22text%22%3A%20%22Details%22%2C%0A%09%09%09%09%22emoji%22%3A%20true%0A%09%09%09%7D%2C%0A%09%09%09%7B%0A%09%09%09%09%22type%22%3A%20%22plain_text%22%2C%0A%09%09%09%09%22text%22%3A%20%22%20%22%2C%0A%09%09%09%09%22emoji%22%3A%20true%0A%09%09%09%7D%2C%0A%09%09%09%7B%0A%09%09%09%09%22type%22%3A%20%22plain_text%22%2C%0A%09%09%09%09%22text%22%3A%20%22blah%22%2C%0A%09%09%09%09%22emoji%22%3A%20true%0A%09%09%09%7D%2C%0A%09%09%09%7B%0A%09%09%09%09%22type%22%3A%20%22plain_text%22%2C%0A%09%09%09%09%22text%22%3A%20%22%20%22%2C%0A%09%09%09%09%22emoji%22%3A%20true%0A%09%09%09%7D%0A%09%09%5D%0A%09%7D%2C%0A%09%7B%0A%09%09%22type%22%3A%20%22section%22%2C%0A%09%09%22fields%22%3A%20%5B%0A%09%09%09%7B%0A%09%09%09%09%22type%22%3A%20%22plain_text%22%2C%0A%09%09%09%09%22text%22%3A%20%22More%20Details%22%2C%0A%09%09%09%09%22emoji%22%3A%20true%0A%09%09%09%7D%2C%0A%09%09%09%7B%0A%09%09%09%09%22type%22%3A%20%22plain_text%22%2C%0A%09%09%09%09%22text%22%3A%20%22%20%22%2C%0A%09%09%09%09%22emoji%22%3A%20true%0A%09%09%09%7D%2C%0A%09%09%09%7B%0A%09%09%09%09%22type%22%3A%20%22plain_text%22%2C%0A%09%09%09%09%22text%22%3A%20%22blah%22%2C%0A%09%09%09%09%22emoji%22%3A%20true%0A%09%09%09%7D%2C%0A%09%09%09%7B%0A%09%09%09%09%22type%22%3A%20%22plain_text%22%2C%0A%09%09%09%09%22text%22%3A%20%22%20%22%2C%0A%09%09%09%09%22emoji%22%3A%20true%0A%09%09%09%7D%2C%0A%09%09%09%7B%0A%09%09%09%09%22type%22%3A%20%22plain_text%22%2C%0A%09%09%09%09%22text%22%3A%20%22blah%22%2C%0A%09%09%09%09%22emoji%22%3A%20true%0A%09%09%09%7D%0A%09%09%5D%0A%09%7D%2C%0A%09%7B%0A%09%09%22type%22%3A%20%22section%22%2C%0A%09%09%22fields%22%3A%20%5B%0A%09%09%09%7B%0A%09%09%09%09%22type%22%3A%20%22plain_text%22%2C%0A%09%09%09%09%22text%22%3A%20%22More%20More%20Details%22%2C%0A%09%09%09%09%22emoji%22%3A%20true%0A%09%09%09%7D%2C%0A%09%09%09%7B%0A%09%09%09%09%22type%22%3A%20%22plain_text%22%2C%0A%09%09%09%09%22text%22%3A%20%22%20%22%2C%0A%09%09%09%09%22emoji%22%3A%20true%0A%09%09%09%7D%2C%0A%09%09%09%7B%0A%09%09%09%09%22type%22%3A%20%22plain_text%22%2C%0A%09%09%09%09%22text%22%3A%20%22blah%22%2C%0A%09%09%09%09%22emoji%22%3A%20true%0A%09%09%09%7D%2C%0A%09%09%09%7B%0A%09%09%09%09%22type%22%3A%20%22plain_text%22%2C%0A%09%09%09%09%22text%22%3A%20%22%20%22%2C%0A%09%09%09%09%22emoji%22%3A%20true%0A%09%09%09%7D%2C%0A%09%09%09%7B%0A%09%09%09%09%22type%22%3A%20%22plain_text%22%2C%0A%09%09%09%09%22text%22%3A%20%22blah%22%2C%0A%09%09%09%09%22emoji%22%3A%20true%0A%09%09%09%7D%2C%0A%09%09%09%7B%0A%09%09%09%09%22type%22%3A%20%22plain_text%22%2C%0A%09%09%09%09%22text%22%3A%20%22%20%22%2C%0A%09%09%09%09%22emoji%22%3A%20true%0A%09%09%09%7D%0A%09%09%5D%0A%09%7D%2C%0A%09%7B%0A%09%09%22type%22%3A%20%22context%22%2C%0A%09%09%22elements%22%3A%20%5B%0A%09%09%09%7B%0A%09%09%09%09%22type%22%3A%20%22plain_text%22%2C%0A%09%09%09%09%22emoji%22%3A%20true%2C%0A%09%09%09%09%22text%22%3A%20%22%5Bblahblah.com%5D%22%0A%09%09%09%7D%0A%09%09%5D%0A%09%7D%2C%0A%09%7B%0A%09%09%22type%22%3A%20%22divider%22%0A%09%7D%2C%0A%09%7B%0A%09%09%22type%22%3A%20%22section%22%2C%0A%09%09%22text%22%3A%20%7B%0A%09%09%09%22type%22%3A%20%22mrkdwn%22%2C%0A%09%09%09%22text%22%3A%20%22Action%3A%22%0A%09%09%7D%2C%0A%09%09%22accessory%22%3A%20%7B%0A%09%09%09%22type%22%3A%20%22static_select%22%2C%0A%09%09%09%22placeholder%22%3A%20%7B%0A%09%09%09%09%22type%22%3A%20%22plain_text%22%2C%0A%09%09%09%09%22emoji%22%3A%20true%2C%0A%09%09%09%09%22text%22%3A%20%22Select%20an%20option%22%0A%09%09%09%7D%2C%0A%09%09%09%22options%22%3A%20%5B%0A%09%09%09%09%7B%0A%09%09%09%09%09%22text%22%3A%20%7B%0A%09%09%09%09%09%09%22type%22%3A%20%22plain_text%22%2C%0A%09%09%09%09%09%09%22emoji%22%3A%20true%2C%0A%09%09%09%09%09%09%22text%22%3A%20%22Approve%22%0A%09%09%09%09%09%7D%2C%0A%09%09%09%09%09%22value%22%3A%20%22-approve%22%0A%09%09%09%09%7D%2C%0A%09%09%09%09%7B%0A%09%09%09%09%09%22text%22%3A%20%7B%0A%09%09%09%09%09%09%22type%22%3A%20%22plain_text%22%2C%0A%09%09%09%09%09%09%22emoji%22%3A%20true%2C%0A%09%09%09%09%09%09%22text%22%3A%20%22Block%20-%20Fraud%22%0A%09%09%09%09%09%7D%2C%0A%09%09%09%09%09%22value%22%3A%20%22-block_fraud%22%0A%09%09%09%09%7D%2C%0A%09%09%09%09%7B%0A%09%09%09%09%09%22text%22%3A%20%7B%0A%09%09%09%09%09%09%22type%22%3A%20%22plain_text%22%2C%0A%09%09%09%09%09%09%22emoji%22%3A%20true%2C%0A%09%09%09%09%09%09%22text%22%3A%20%22Block%20-%20Wrong%20Info%22%0A%09%09%09%09%09%7D%2C%0A%09%09%09%09%09%22value%22%3A%20%22-block_wrong_info%22%0A%09%09%09%09%7D%0A%09%09%09%5D%2C%0A%09%09%09%22action_id%22%3A%20%22escalation_response%22%0A%09%09%7D%0A%09%7D%0A%5D)
2. Send to interactive message app
3. Middleware can't filter properly for this message

> slackInteractions.action({'action_id': "escalation_response"}, escalationResponseHandler)

#### Expected result:

I expect middleware to be able to detect it when I use action id as a filter. The version from to 1.1.0 did that successfully. This version does not.

#### Actual result:

What actually happened, middleware skips this.

#### Request:

1) How do I fix my middleware to be able to listen to the messages I have in the block kit?
2) How do I set up a catch all filter so I can record all unrelated events?
3) Could the slack team add me to the slack team where I Can message them directly please? Customer support was not helpful.

**Comment by shanedewael** [comment]
@awcchungster Sorry to hear you ran into this issue. I'm not quite sure how your listener was working up until `1.1.0` but you should be able to fix your issue by changing `action_id` to `actionId`. Using Block Kit-supported listeners is documented on the README and you can find examples relevant to your listener at https://github.com/slackapi/node-slack-interactive-messages#user-content-action-matching-with-blocks.

If you run into any more issues, let me know

**Comment by awcchungster** [comment]
@shanedewael 

Changed to: 

> slackInteractions.action({'actionId': "escalation_response"}, escalationResponseHandler)

Still not working.

**Comment by shanedewael** [comment]
It should be 

> slackInteractions.action({actionId: "escalation_response"}, escalationResponseHandler)

(no quotes)

**Comment by shanedewael** [comment]
Actually, I don't think the change above has an effect since Javascript will treat `'actionId'` and `actionId` the same when it's in an object.

That being said, it might be helpful to turn on debugging to try and get to the bottom of what's happening. To enable it for the entire package, you can add `DEBUG=@slack/interactive-messages:*` when you're running the app from the command line. So if you previously ran your app using `node app.js`, you would use the command `DEBUG=@slack/interactive-messages:* node app.js`. This is talked about in [the maintainers guide](https://github.com/slackapi/node-slack-interactive-messages/blob/master/.github/maintainers_guide.md).

On our opensource projects, we don't typically offer 1-on-1 support and do our best to answer any open Github issues in a timely manner.

**Comment by awcchungster** [comment]
>   @slack/interactive-messages:http-handler request received - method: POST, path: / +0ms
>   @slack/interactive-messages:http-handler request signature is not valid +2ms

**Comment by shanedewael** [comment]
Ah, that error makes sense.

You'll want to make sure you're passing in a valid `Signing Secret` from your app configuration page
![image](https://user-images.githubusercontent.com/3411005/53669277-69ee5c80-3c2b-11e9-814e-b45f58c55987.png)

You pass the signing secret into the app when you instantiate the adapter:
```
// Create the adapter using the app's signing secret, read from environment variable
const slackInteractions = createMessageAdapter(YOUR_SIGNING_SECRET_HERE);
```

**Comment by awcchungster** [comment]
Okay, hmm interesting, somehow the keys got changed. Thank you for the clarification.

> escalationResponseHandler = (payload, respond) => {
> 
>         // for help, refer to this documentation page: 
>         // https://github.com/slackapi/node-slack-interactive-messages
> 
>         let message_immediate_response = payload.message
>         ....
>         return message_immediate_response

I return message_immediate_response to update the post, and then the respond() is used in my callback after the steps are fully processed. The return message_immediate_response isn't changing anything. Any thoughts of what I'm missing?

**Comment by aoberoi** [comment]
@awcchungster we just released v1.1.1 of this package yesterday, which a pretty critical fix included. i'm not sure whether this would have affected your app in the way you're asking about here, but its possible. can you give it a try with the latest release and see if its still happening?

if for some reason your app takes longer than 2.5 seconds to (synchronously) process the message before returning in the callback, then you might find that the `return message_immediate_response` has no effect. if you shared the debug logs, we would be able to figure out if that was happening. also, you might be able to look at them yourself and trace any problems.
